File: C#/T1/Project/PetSimulator/Manager.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace PetSimulator
{
    class Manager
    {
        private static int dayCount = 1;

        enum Days
        {
            Monday,
            Tuesday,
            Wednesday,
            Thursday,
            Friday,
            Saturday,
            Sunday
        }

        private static int i = 0;
        private static int count = 0;
        public static void checkDay()
        {
            if (count != 2)
            {
                count++;
            }
            else if (count == 2)
            {
                i++;
                count = 0;
            }

            Console.ForegroundColor = ConsoleColor.Yellow;
            Console.Write("[GAME] ");
            Console.ResetColor();
            Console.Write("Day ");
            Console.ForegroundColor = ConsoleColor.Red;
            Console.Write("{0}\n", dayCount);
            Console.ResetColor();

            Days days = (Days)i;
            switch (days)
            {
                case Days.Monday:
                    Console.ForegroundColor = ConsoleColor.Yellow;
                    Console.Write("[GAME] ");
                    Console.ResetColor();
                    Console.Write("Today is ");
                    Console.ForegroundColor = ConsoleColor.DarkYellow;
                    Console.Write("{0}\n", days);
                    Console.ResetColor();
                    break;
                case Days.Tuesday:
                    Console.ForegroundColor = ConsoleColor.Yellow;
                    Console.Write("[GAME] ");
                    Console.ResetColor();
                    Console.Write("Today is ");
                    Console.ForegroundColor = ConsoleColor.DarkMagenta;
                    Console.Write("{0}\n", days);
                    Console.ResetColor();
                    break;
                case Days.Wednesday:
                    Console.ForegroundColor = ConsoleColor.Yellow;
                    Console.Write("[GAME] ");
                    Console.ResetColor();
                    Console.Write("Today is ");
                    Console.ForegroundColor = ConsoleColor.DarkGreen;
                    Console.Write("{0}\n", days);
                    Console.ResetColor();
                    break;
                case Days.Thursday:
                    Console.ForegroundColor = ConsoleColor.Yellow;
                    Console.Write("[GAME] ");
                    Console.ResetColor();
                    Console.Write("Today is ");
                    Console.ForegroundColor = ConsoleColor.DarkRed;
                    Console.Write("{0}\n", days);
                    Console.ResetColor();
                    break;
                case Days.Friday:
                    Console.ForegroundColor = ConsoleColor.Yellow;
                    Console.Write("[GAME] ");
                    Console.ResetColor();
                    Console.Write("Today is ");
                    Console.ForegroundColor = ConsoleColor.Cyan;
                    Console.Write("{0}\n", days);
                    Console.ResetColor();
                    break;
                case Days.Saturday:
                    Console.ForegroundColor = ConsoleColor.Yellow;
                    Console.Write("[GAME] ");
                    Console.ResetColor();
                    Console.Write("Today is ");
                    Console.ForegroundColor = ConsoleColor.Magenta;
                    Console.Write("{0}\n", days);
                    Console.ResetColor();
                    break;
                case Days.Sunday:
                    Console.ForegroundColor = ConsoleColor.Yellow;
                    Console.Write("[GAME] ");
                    Console.ResetColor();
                    Console.Write("Today is ");
                    Console.ForegroundColor = ConsoleColor.Red;
                    Console.Write("{0}\n", days);
                    Console.ResetColor();
                    break;
            }

            dayCount++;
        }

        public static void showWelcomeScreen()
        {
            Console.ForegroundColor = ConsoleColor.Blue;
            Console.WriteLine("=========================================================================\n\n\n");
            Console.ForegroundColor = ConsoleColor.Red;
            Console.WriteLine("               Welcome to Tamagottagettemall Pet Simulator              \n\n\n\n\n");
            Console.ForegroundColor = ConsoleColor.Green;
            Console.WriteLine("                     Press any key to start the game                    \n\n");
            Console.ForegroundColor = ConsoleColor.Blue;
            Console.WriteLine("=========================================================================\n");
            Console.ResetColor();
            Console.ReadLine();
        }

        public static void showEndScreen()
        {
            Console.Clear();
            Console.ForegroundColor = ConsoleColor.Blue;
            Console.WriteLine("=========================================================================\n\n\n");
            Console.ForegroundColor = ConsoleColor.Red;
            Console.WriteLine("                                Good Bye!              \n\n\n\n\n");
            Console.ForegroundColor = ConsoleColor.Green;
            Console.WriteLine("                     Press any key to exit the game                    \n\n");
            Console.ForegroundColor = ConsoleColor.Blue;
            Console.WriteLine("=========================================================================\n");
            Console.ResetColor();
            Console.ReadLine();
        }

        public static void showDeathScreen(Pet p)
        {
            Console.ForegroundColor = ConsoleColor.Yellow;
            Console.Write("[GAME] ");
            Console.ResetColor();
            Console.Write("{0} Died!\n", p.petName);
        }

        public static void showResetPetScreen(Pet p)
        {
            Console.ForegroundColor = ConsoleColor.Yellow;
            Console.Write("[GAME] ");
            Console.ResetColor();
            Console.Write("{0}'s stats has been reset\n", p.petName);
            p.Reset();
        }

        public static void showCommandScreen(Player player, Pet pet)
        {
            Console.ForegroundColor = ConsoleColor.Yellow;
            Console.Write("[GAME] ");
            Console.ResetColor();
            Console.Write("Hi {0}, Welcome to the game\n", player.playerName);
            Console.ForegroundColor = ConsoleColor.Yellow;
            Console.Write("[GAME] ");
            Console.ResetColor();
            Console.Write("What would you like to do today?\n");
            Console.WriteLine("1. Feed {0}", pet.petName);
            Console.WriteLine("2. Walk {0}", pet.petName);
            Console.WriteLine("3. Sleep {0}", pet.petName);
            Console.WriteLine("4. Play {0}", pet.petName);
            Console.WriteLine("5. Movie {0}", pet.petName);
            Console.WriteLine("6. Toilet {0}", pet.petName);
        }

        public static void showPetStats(Pet pet)
        {
            Console.ForegroundColor = ConsoleColor.Magenta;
            Console.Write("[{0}] ", pet.petName);
            Console.ResetColor();
            Console.Write("Stats:\n");
            Console.WriteLine("Hunger: {0}", pet.hunger);
            Console.WriteLine("Thirst: {0}", pet.thirst);
            Console.WriteLine("Sleepiness: {0}", pet.sleepiness);
            Console.WriteLine("Happiness: {0}", pet.happiness);
            Console.Write("\n");
        }

        public static void showPauseScreen()
        {
            Console.WriteLine("Press any key to continue.");
            Console.ReadLine();
        }

        public static bool checkPet(Pet p)
        {
            return p.checkHealth();
        }

        public static void askPlayerName(Player p)
        {
            Console.ForegroundColor = ConsoleColor.Yellow;
            Console.Write("[GAME] ");
            Console.ResetColor();
            Console.Write("Enter Player Name: ");
            p.playerName = Console.ReadLine();
        }

        public static void askPetName(Pet p)
        {
            Console.ForegroundColor = ConsoleColor.Yellow;
            Console.Write("[GAME] ");
            Console.ResetColor();
            Console.Write("Enter Pet Name: ");
            p.petName = Console.ReadLine();
        }

        public static void askForCommand(Player player, Pet pet)
        {
            Console.ForegroundColor = ConsoleColor.Yellow;
            Console.Write("[GAME] ");
            Console.ResetColor();
            Console.Write("Enter your choice: ");
            string choice = Console.ReadLine();

            Command.checkCommands(choice, player, pet);
        }

        public static bool askToPlayAgain()
        {
            Console.ForegroundColor = ConsoleColor.Yellow;
            Console.Write("[GAME] ");
            Console.ResetColor();
            Console.Write("Do you want to play again?\n");
            Console.WriteLine("Y - Yes | N - No");

            Console.ForegroundColor = ConsoleColor.Yellow;
            Console.Write("[GAME] ");
            Console.ResetColor();
            Console.Write("Enter your choice: ");
            string decision = Console.ReadLine();

            if (decision == "Y" || decision == "y")
            {
                return true;
            }
            else
            {
                return false;
            }
        }
    }
}

File: C++/T2/PetSimulator/PetSimulator/Manager.cpp
#include <iostream>

#include "ConsoleColor.h"
#include "Manager.h"

using namespace std;

Manager::Manager() :
	counter(0),
	dayCount(0),
	totalDayCount(1),
	command(nullptr),
	player(nullptr),
	pet(nullptr)
{}

Manager::~Manager()
{}

void Manager::Initialize()
{
	command = new Command();
	pet = new Pet();
	player = new Player();
}

void Manager::Terminate()
{
	delete command;
	delete pet;
	delete player;
}

void Manager::showWelcomeScreen()
{
	ConsoleColor cc;
	cc.Blue();
	cout << "=========================================================================\n\n\n";
	cc.Red();
	cout << "               Welcome to Tamagottagettemall Pet Simulator              \n\n\n\n\n";
	cc.Green();
	cout << "                     Press any key to start the game                    \n\n";
	cc.Blue();
	cout << "=========================================================================\n";
	cc.White();
	system("pause");
}

void Manager::showEndScreen()
{
	ConsoleColor cc;
	cc.Blue();
	cout << "=========================================================================\n\n\n";
	cc.Red();
	cout << "                                Good Bye!              \n\n\n\n\n";
	cc.Green();
	cout << "                     Press any key to exit the game                    \n\n";
	cc.Blue();
	cout << "=========================================================================\n";
	cc.White();
	system("pause");
}

void Manager::showDeathScreen()
{
	ConsoleColor cc;
	cc.Yellow();
	cout << "[GAME] ";
	cc.White();
	cout << pet->getName() << " Died!\n";
}

void Manager::showResetScreen()
{
	ConsoleColor cc;
	cc.Yellow();
	cout << "[GAME] ";
	cc.White();
	cout << pet->getName() << " stats has been reset\n";
	
	// Reset all stats
	pet->Reset();
	counter = 0;
	dayCount = 0;
	totalDayCount = 1;
}

void Manager::showCommandScreen()
{
	ConsoleColor cc;
	cc.Yellow();
	cout << "[GAME] ";
	cc.White();
	cout << "Hi " << player->getName() << ", Welcome to the game\n";
	cc.Yellow();
	cout << "[GAME] ";
	cc.White();
	cout << "What would you like to do today?\n";
	cout << "1. Feed " << pet->getName() << endl;
	cout << "2. Walk " << pet->getName() << endl;
	cout << "3. Sleep " << pet->getName() << endl;
	cout << "4. Play " << pet->getName() << endl;
	cout << "5. Movie " << pet->getName() << endl;
	cout << "6. Toilet " << pet->getName() << endl;
}

void Manager::showPetStats()
{
	ConsoleColor cc;
	cc.Magenta();
	cout << "[" << pet->getName() << "] ";
	cc.White();
	cout << "Stats: \n";
	cout << "Hunger: " << pet->getHunger() << endl;
	cout << "Thirst: " << pet->getThirst() << endl;
	cout << "Sleepiness: " << pet->getSleepiness() << endl;
	cout << "Happiness: " << pet->getHappiness() << endl;
	cout << endl;
}

// Keep track of day
void Manager::checkDay()
{
	if (counter != 2)
	{
		counter++;
	}
	else if (counter == 2)
	{
		dayCount++;
		counter = 0;
	}

	if (dayCount > 6)
	{
		dayCount = 0;
	}

	ConsoleColor cc;
	cc.Yellow();
	cout << "[Game] ";
	cc.White();
	cout << "Day ";
	cc.Red();
	cout << totalDayCount << endl;
	cc.White();

	// Print current day
	switch (dayCount)
	{
	case Monday:
		cc.Yellow();
		cout << "[Game] ";
		cc.White();
		cout << "Today is ";
		cc.DarkYellow();
		cout << "Monday\n";
		cc.White();
		break;
	case Tuesday:
		cc.Yellow();
		cout << "[Game] ";
		cc.White();
		cout << "Today is ";
		cc.DarkMagenta();
		cout << "Tuesday\n";
		cc.White();
		break;
	case Wednesday:
		cc.Yellow();
		cout << "[Game] ";
		cc.White();
		cout << "Today is ";
		cc.DarkGreen();
		cout << "Wednesday\n";
		cc.White();
		break;
	case Thursday:
		cc.Yellow();
		cout << "[Game] ";
		cc.White();
		cout << "Today is ";
		cc.DarkRed();
		cout << "Thursday\n";
		cc.White();
		break;
	case Friday:
		cc.Yellow();
		cout << "[Game] ";
		cc.White();
		cout << "Today is ";
		cc.Cyan();
		cout << "Cyan\n";
		cc.White();
		break;
	case Saturday:
		cc.Yellow();
		cout << "[Game] ";
		cc.White();
		cout << "Today is ";
		cc.Magenta();
		cout << "Saturday\n";
		cc.White();
		break;
	case Sunday:
		cc.Yellow();
		cout << "[Game] ";
		cc.White();
		cout << "Today is ";
		cc.Red();
		cout << "Sunday\n";
		cc.White();
		break;
	}

	totalDayCount++;
}

bool Manager::checkPet()
{
	return pet->checkHealth();
}

void Manager::askPlayerName()
{
	ConsoleColor cc;
	cc.Yellow();
	cout << "[Game] ";
	cc.White();
	cout << "Enter Player Name: ";

	std::string name;
	cin >> name;
	player->setName(name);
}

void Manager::askPetName()
{
	ConsoleColor cc;
	cc.Yellow();
	cout << "[Game] ";
	cc.White();
	cout << "Enter Pet Name: ";

	std::string name;
	cin >> name;
	pet->setName(name);
}

void Manager::askForCommand()
{
	ConsoleColor cc;
	cc.Yellow();
	cout << "[Game] ";
	cc.White();
	cout << "Enter your choice: ";

	int choice;
	cin >> choice;
	
	command->checkCommand(choice, player, pet);
}

bool Manager::askToPlayAgain()
{
	ConsoleColor cc;
	cc.Yellow();
	cout << "[Game] ";
	cc.White();
	cout << "Do you want to play again?\n";
	cout << "Y - Yes | N - No\n";

	cc.Yellow();
	cout << "[Game] ";
	cc.White();
	cout << "Enter your choice: ";

	char choice;
	cin >> choice;
	
	if (choice == 'Y' || choice == 'y')
	{
		return true;
	}
	else
	{
		return false;
	}
}
